components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/oauth/OAuthAuthenticator.java

```diff
@@ -242,6 +292,20 @@ else if(apiVersion == null){
             return new AuthenticationResponse(false, isMandatory, true,
                     APISecurityConstants.API_AUTH_MISSING_CREDENTIALS, "Required OAuth credentials not provided");
         } else {
+            //Start JWT token validation
+            if (isJwtToken) {
+                try {
+                    AuthenticationContext authenticationContext = jwtValidator.authenticate(apiKey, synCtx, openAPI);
+                    APISecurityUtils.setAuthenticationContext(synCtx, authenticationContext, securityContextHeader);
+                    log.debug("User is authorized using JWT token to access the resource.");
```

Comment:
improve the debug log by appending specific information such as user etc

Reply:
Fixed in https://github.com/wso2/carbon-apimgt/pull/9250